src/lib/scheduler/abstract_task.hpp

```diff
@@ -170,6 +170,16 @@ class AbstractTask : public std::enable_shared_from_this<AbstractTask> {
    */
   bool try_mark_as_assigned_to_worker();
 
+  /**
+   * Sets the shutdown flag that signals workers to shut down.
```

Comment:
Could you elaborate when, why an by whom this flag should be set? To quote a random dude I met somewhere (don't remember his name though): "I don't wanna look at another header file to understand sth."